source/kernel/src/registry/mod.rs

```diff
@@ -284,26 +344,44 @@ pub struct KernelHandle<RD: RegisteredDriver> {
     request_ctr: u32,
 }
 
-type ErasedDeserHandler = unsafe fn(
+type ErasedReqDeser = unsafe fn(
     UserRequest<'_>,
     &ErasedKProducer,
     &bbq::MpscProducer,
     ServiceId,
     ClientId,
 ) -> Result<(), UserHandlerError>;
 
+type ErasedHandshake = unsafe fn(
```

Comment:
Honestly, if we want to just yank all of the userspace stuff for now, we've already removed the k2u/u2k rings, and this will all be redesigned to be interface-like later.

Reply:
my motivation for actually writing the userspace code was mostly that i wanted to validate that the new design didn't completely break our type erasure mechanism, and that it was possible to do something like this for a serialization based service. i'm happy to remove this code and add something similar back in later, but i have a slight preference for keeping it so that we can build on top of it when integrating the new serializing channel. fundamentally up to you, though.